pkg/sidecar/k8s_network.go

```diff
@@ -33,18 +36,92 @@ type K8sNetwork struct {
 	cninet          *libcni.CNIConfig
 	subnet          string
 	netnsPath       string
+	initialized     bool
 }
 
 func (n *K8sNetwork) Close() error {
 	n.nl.Delete()
 	return nil
 }
 
+// getExistingIpRange returns an IP range, if it exists and is attached, that contains the
+// given matchingAddr
+func getExistingIpRange(matchingAddr string) *net.IPNet {
+	// remove mask from matching Addr
+	matchingAddr = strings.Split(matchingAddr, "/")[0]
+
+	logging.S().Infof("Checking local ranges for address %s", matchingAddr)
+
+	addrs, err := net.InterfaceAddrs()
+	if err != nil {
+		logging.S().Errorf("interfaceAddrs error: %+v\n", err.Error())
+		return nil
+	}
+
+	if len(addrs) == 0 {
+		logging.S().Warnf("No InterfaceAddrs exist")
+	} else {
+		// check all interface addresses
+		for _, i := range addrs {
+			_, parsedIp, err := net.ParseCIDR(i.String())
+			if err != nil {
+				logging.S().Errorf(err.Error())
+				continue
+			}
+			iNet := &ptypes.IPNet{IPNet: *parsedIp}
+			matchingNetIp := net.ParseIP(matchingAddr)
+			// does the range contain the target IP?
+			if iNet.Contains(matchingNetIp) {
+				return parsedIp
+			}
+		}
+	}
+	return nil
+
+}
+
+// InitializeNetwork initializes the k8s network. This should be once, before any additional configuration takes place
+func (n *K8sNetwork) InitializeNetwork(ctx context.Context) error {
+	// It is possible an existing address is already assigned by the k8s CNI, and the network doesn't track it
```

Comment:
See here: https://github.com/testground/testground/pull/1350/files#r984635878

It will actually happen 100% of the time. I will refactor the comment to reflect this.

Reply:
Please do not resolve conversations for me. It makes it easier for me to miss your answers.